universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramReviewController.java

```diff
@@ -0,0 +1,131 @@
+package controllers.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.ProfileUtils;
+import com.google.common.collect.ImmutableList;
+import controllers.CiviFormController;
+import java.util.Optional;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import org.pac4j.play.java.Secure;
+import play.libs.concurrent.HttpExecutionContext;
+import play.mvc.Call;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import repository.ApplicationRepository;
+import services.applicant.AnswerData;
+import services.applicant.ApplicantService;
+import services.program.ProgramNotFoundException;
+import views.applicant.ApplicantProgramSummaryView;
+
+/**
+ * Controller for reviewing program responses for an applicant.
+ *
+ * <p>CAUTION: You must explicitly check the current profile so that an unauthorized user cannot
```

Comment:
question (non-blocking): Could this be done at the routing level (maybe in addition to what's done in this file)?